client/src/app/pages/identities/identities.tsx

```diff
@@ -223,11 +226,17 @@ export const Identities: React.FC = () => {
         {
           title: (
             <AppTableActionButtons
-              isDeleteEnabled={trackers.some(
-                (tracker) => tracker?.identity?.id === item.id
-              )}
+              isDeleteEnabled={
+                trackers.some((tracker) => tracker?.identity?.id === item.id) ||
+                applications?.some(
+                  (app) => app?.identities?.some((id) => id.id === item.id)
+                ) ||
+                targets?.some(
+                  (target) => target?.ruleset?.identity?.id === item.id
+                )
+              }
               tooltipMessage={
-                "Cannot delete credential assigned to a JIRA tracker."
+                "Cannot delete credential as it is currently in use."
```

Comment:
Suggested improvement: Add the name of the thing that is using the credential so it is easy to find.

"Credential is in use by application Foxtrot Oscar and cannot be deleted."